tests/MongoDB.Driver.Legacy.Tests/Operations/BulkWriteOperationTests.cs

```diff
@@ -1252,6 +1274,23 @@ private void CheckExpectedResult(ExpectedResult expectedResult, BulkWriteResult<
             }
         }
 
+        private MongoCollection<BsonDocument> GetCollection(MongoServer server)
+        {
+            return server
+                .GetDatabase(CoreTestConfiguration.DatabaseNamespace.DatabaseName)
+                .GetCollection(GetType().Name);
+        }
+
+        private MongoServer GetServerFromDisposableClient()
+        {
+            DisposableMongoClient client = DriverTestConfiguration.CreateDisposableClient(
+                settings =>
+                {
+                    settings.RetryWrites = false;
+                });
+            return client.GetServer();
+        }
```

Comment:
I don't think we should be using DisposableMongoClient in this context (we should only use one when we intend to Dispose it).

Remove this method, and instead let's add the following method to LegacyTestConfiguration.cs instead:

            public static MongoServer GetServer(
                bool? retryReads = null,
                bool? retryWrites = null)
            {
                var settings = DriverTestConfiguration.GetClientSettings();
                if (retryReads.HasValue)
                {
                    settings.RetryReads = retryReads.Value;
                }
                if (retryWrites.HasValue)
                {
                    settings.RetryWrites = retryWrites.Value;
                }
                var client = new MongoClient(settings);
    #pragma warning disable 618
                return client.GetServer();
    #pragma warning restore 618
            }

Note: you could simplify it a bit by removing retryReads since only retryWrites is needed so far.

Reply:
Done